paasta_tools/envoy_tools.py

```diff
@@ -49,10 +47,48 @@
 )
 
 
+def are_services_up_in_pod(
+    envoy_host: str,
+    envoy_admin_port: int,
+    envoy_admin_uri_format: str,
+    registrations: Collection[str],
+    pod_ip: str,
+    pod_port: int,
+) -> bool:
+    """Returns whether a service in a k8s pod is reachable via envoy
+    :param envoy_host: The host that this check should contact for replication information.
+    :param envoy_admin_port: The port that Envoy's admin interface is listening on
+    :param registrations: The service_name.instance_name of the services
+    :param pod_ip: IP of the pod itself
+    :param pod_port: The port to reach the service in the pod
+    """
+    services_with_atleast_one_backend_up = {
+        registration: False for registration in registrations
+    }
+
+    for registration in registrations:
+        backends = get_backends(
+            registration,
+            envoy_host=envoy_host,
+            envoy_admin_port=envoy_admin_port,
+            envoy_admin_uri_format=envoy_admin_uri_format,
+        )
+
+        for be in backends:
+            if (
+                be[0]["eds_health_status"] == "HEALTHY"
+                and be[0]["address"] == pod_ip
+                and be[0]["port_value"] == pod_port
+            ):
+                services_with_atleast_one_backend_up[registration] = True
```

Comment:
IMO `services_with_atleast_one_backend_up` makes the code unnecessary complicated. 🙂 
Maybe `return False` here, `return True` instead of `return all(services_with_atleast_one_backend_up.values())` below, and get rid of `services_with_atleast_one_backend_up`?

Reply:
You are right. I guess this approach (using `services_with_atleast_one_backend_up`) made sense on smartstack, as synapse [would often keep](https://github.com/Yelp/paasta/blob/master/paasta_tools/smartstack_tools.py#L393) old backends around for a while.
But I don't think this is needed on Envoy, so I'll go with your recommendation :)